test/e2e/useraccount_test.go

```diff
@@ -0,0 +1,157 @@
+	require.NoError(t, err, "failed to initialize cluster resources")
+	t.Log("Initialized cluster resources")
+
+	namespace, err := ctx.GetNamespace()
+	require.NoError(t, err, "failed to get namespace where operator needs to run")
+
+	// get global framework variables
+	f := framework.Global
+	client := f.Client.Client
+
+	// wait for operator to be ready
+	err = e2eutil.WaitForDeployment(t, f.KubeClient, namespace, "member-operator", 1, retryInterval, timeout)
+	require.NoError(t, err, "failed while waiting for operator deployment")
+
+	t.Log("member-operator is ready and running state")
+
+	extraUserAcc := createUserAccount(t, f, ctx, "amar")
+	t.Log("extra useraccount created at start")
+
+	// create useraccount
+	userAcc := newUserAcc(t, f, ctx, "johnsmith")
+	err = f.Client.Create(context.TODO(), userAcc, &framework.CleanupOptions{TestContext: ctx, Timeout: cleanupTimeout, RetryInterval: cleanupRetryInterval})
+	require.NoError(t, err)
+	t.Logf("user account '%s' created", userAcc.Name)
+
+	t.Run("verify_useraccount_ok", func(t *testing.T) {
+		err := verifyResources(t, f, userAcc)
+		assert.NoError(t, err)
+	})
+
+	t.Run("delete_user_ok", func(t *testing.T) {
+		user := &userv1.User{}
+		err := client.Get(context.TODO(), types.NamespacedName{Name: userAcc.Name}, user)
+		require.NoError(t, err)
+
+		err = client.Delete(context.TODO(), user)
+		require.NoError(t, err)
+
+		err = verifyResources(t, f, userAcc)
+		assert.NoError(t, err)
+	})
+
+	t.Run("delete_identity_ok", func(t *testing.T) {
+		identity := &userv1.Identity{}
+		err := client.Get(context.TODO(), types.NamespacedName{Name: useraccount.ToIdentityName(userAcc.Spec.UserID)}, identity)
+		require.NoError(t, err)
+
+		err = client.Delete(context.TODO(), identity)
+		require.NoError(t, err)
+
+		err = verifyResources(t, f, userAcc)
+		assert.NoError(t, err)
+	})
+
+	t.Run("delete_user_mapping_ok", func(t *testing.T) {
+		user := &userv1.User{}
+		err := client.Get(context.TODO(), types.NamespacedName{Name: userAcc.Name}, user)
+		require.NoError(t, err)
+
+		user.Identities = []string{}
+		err = client.Update(context.TODO(), user)
+		require.NoError(t, err)
+
+		err = verifyResources(t, f, userAcc)
+		assert.NoError(t, err)
+	})
+
+	t.Run("delete_identity_mapping_ok", func(t *testing.T) {
+		identity := &userv1.Identity{}
+		err := client.Get(context.TODO(), types.NamespacedName{Name: useraccount.ToIdentityName(userAcc.Spec.UserID)}, identity)
+		require.NoError(t, err)
+
+		identity.User = corev1.ObjectReference{Name: "", UID: ""}
+		err = client.Update(context.TODO(), identity)
+		require.NoError(t, err)
+
+		err = verifyResources(t, f, userAcc)
+		assert.NoError(t, err)
+	})
+
+	err = verifyResources(t, f, extraUserAcc)
+	assert.NoError(t, err)
+	t.Log("extra useraccount verified at end")
+}
+
+func newUserAcc(t *testing.T, f *framework.Framework, ctx *framework.TestCtx, name string) *toolchainv1alpha1.UserAccount {
```

Comment:
I think that the only parameter that is used in the function is the `name`, the rest are redundant

Reply:
done [2520703](https://github.com/codeready-toolchain/member-operator/pull/26/commits/25207039d1264b68479cef26dd207729da7ad5d7)